ts/client/src/client.ts

```diff
@@ -5206,9 +5222,80 @@ export class MangoClient {
         .map((serumPosition) => serumPosition.openOrders),
     );
 
+    const fallbackMap = await this.deriveFallbackOracleContexts(group);
+    const fallbacks: PublicKey[] = [];
+
+    for (const oracle of mintInfos.map((mintInfo) => mintInfo.oracle)) {
+      if (fallbackMap.has(oracle)) {
+        fallbacks.push(...fallbackMap.get(oracle)!);
+      }
+    }
+
+    for (const fallback of uniq(fallbacks)) {
+      if (
+        !healthRemainingAccounts.find((h) => h.equals(fallback)) &&
+        !fallback.equals(PublicKey.default)
+      ) {
+        healthRemainingAccounts.push(fallback);
+      }
+    }
     return healthRemainingAccounts;
   }
 
+  /**This function assumes that the provided group has loaded banks*/
+  public async deriveFallbackOracleContexts(
+    group: Group,
+  ): Promise<Map<PublicKey, [PublicKey, PublicKey]>> {
+    // fixed
+    if (typeof this.fallbackOracleConfig !== 'string') {
+      if (this.fixedFallbacks.size == 0) {
+        const oracles: PublicKey[] = this.fallbackOracleConfig;
+        const fallbacks: PublicKey[] = [];
+        Array.from(group.banksMapByTokenIndex.values()).forEach((b) => {
+          if (oracles.find((o) => o.toBase58() === b[0].oracle.toBase58())) {
+            fallbacks.push(b[0].fallbackOracle);
+          }
+        });
+        this.fixedFallbacks = await createFallbackOracleMap(
+          this.connection,
+          oracles,
+          fallbacks,
+        );
+      }
+      return this.fixedFallbacks;
+    }
+
+    switch (this.fallbackOracleConfig) {
+      case 'never':
+        return new Map();
+      case 'dynamic': {
+        const nowSlot = await this.connection.getSlot();
+        const oracles: PublicKey[] = [];
+        const fallbacks: PublicKey[] = [];
+        await group.reloadBankOraclePrices(this);
+        Array.from(group.banksMapByTokenIndex.values())
+          .filter((b) => b[0].isOracleStaleOrUnconfident(nowSlot))
+          .forEach((b) => {
+            oracles.push(b[0].oracle);
+            fallbacks.push(b[0].fallbackOracle);
+          });
+        return createFallbackOracleMap(this.connection, oracles, fallbacks);
+      }
+      case 'all': {
+        const oracles: PublicKey[] = [];
+        const fallbacks: PublicKey[] = [];
+        Array.from(group.banksMapByTokenIndex.values()).forEach((b) => {
```

Comment:
imo this should only include the fallbacks for the oracles that are used by the mangoAccounts passed to buildHealthRemainingAccounts(). Fallbacks for all listed tokens might just blow the account limit.